src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -2748,6 +2748,33 @@ public override BoundNode VisitConversion(BoundConversion node)
 
             var result = base.VisitConversion(node);
 
+            if (IsConditionalState)
```

Comment:
Feels like this code should go in the AbstractDataFlowPass since it's similar to what we do on branch analysis.